### PR TITLE
Adding test to reference repository

### PR DESCRIPTION
**Pull request by mgonyan**

### CODE REVIEW

**Comment by Ocramius** [comment]
Took me a while to understand what was going on. Can you please re-send the patch without the changes in the original class, and without changing position of the methods? You are just breaking git history/blame functionality that way :-\

File: tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php [hunk 1]
```diff
@@ -142,4 +147,83 @@ public function testUndefinedReference()
 
         $referenceRepository->getReference('foo');
     }
+
+    /**
+     * @expectedException \BadMethodCallException
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Please use `expectException` in the body of the method

File: tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php [hunk 2]
```diff
@@ -142,4 +147,83 @@ public function testUndefinedReference()
...
+     */
+    public function testThrowsExceptionAddingDuplicatedReference()
+    {
+        $referenceRepository = new ReferenceRepository($this->getMockSqliteEntityManager());
+        $referenceRepository->addReference('duplicated_reference', new \stdClass());
+        $referenceRepository->addReference('duplicated_reference', new \stdClass());
+    }
+
+    /**
+     * @expectedException \OutOfBoundsException
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Please use `expectException` in the body of the method

File: tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php [hunk 3]
```diff
@@ -142,4 +147,83 @@ public function testUndefinedReference()
 
         $referenceRepository->getReference('foo');
     }
+
+    /**
+     * @expectedException \BadMethodCallException
+     * @expectedExceptionMessage (duplicated_reference)
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Please use `expectExceptionMessage` in the body of the method

File: tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php [hunk 4]
```diff
@@ -142,4 +147,83 @@ public function testUndefinedReference()
...
+    public function testThrowsExceptionAddingDuplicatedReference()
+    {
+        $referenceRepository = new ReferenceRepository($this->getMockSqliteEntityManager());
+        $referenceRepository->addReference('duplicated_reference', new \stdClass());
+        $referenceRepository->addReference('duplicated_reference', new \stdClass());
+    }
+
+    /**
+     * @expectedException \OutOfBoundsException
+     * @expectedExceptionMessage (missing_reference)
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Please use `expectExceptionMessage` in the body of the method

**Comment by mikeSimonson** [comment]
@mgonyan Thanks

**Pull request merged**